### PR TITLE
python3Packages.aiodiscover: init at 1.3.2

### PR DESCRIPTION
**Pull request by fabaff**
###### Motivation for this change
Python module to discover hosts via ARP and PTR lookup

https://github.com/bdraco/aiodiscover

This will become a Home Assistant dependency.

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [x] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

**Comment by fabaff** [comment]
Result of `nixpkgs-review pr 118176` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>6 packages built:</summary>
  <ul>
    <li>python38Packages.aiodiscover</li>
    <li>python38Packages.async-dns</li>
    <li>python38Packages.pytest-raises</li>
    <li>python39Packages.aiodiscover</li>
    <li>python39Packages.async-dns</li>
    <li>python39Packages.pytest-raises</li>
  </ul>
</details>

File: pkgs/development/python-modules/async-dns/default.nix [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+
+  src = fetchFromGitHub {
+    owner = "gera2ld";
+    repo = "async_dns";
+    rev = "v${version}";
+    sha256 = "0hrpjgbssvr95z9jdnmx853r6h6mih0w138hz301ll94mi8zpbsg";
+  };
+
+  nativeBuildInputs = [
+    # Switch to poetry-core, https://github.com/gera2ld/async_dns/pull/22
```
Comment (SuperSandro2000) [inline pkgs/development/python-modules/async-dns/default.nix:23]: Can we fetch that PR?

File: pkgs/development/python-modules/pytest-raises/default.nix [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+
+  src = fetchFromGitHub {
+    owner = "Lemmons";
+    repo = pname;
+    rev = version;
+    sha256 = "0gbb4kml2qv7flp66i73mgb4qihdaybb6c96b5dw3mhydhymcsy2";
+  };
+
+  propagatedBuildInputs = [
+    pytest
```
Comment (SuperSandro2000) [inline pkgs/development/python-modules/pytest-raises/default.nix:20]: Please don't propagate pytest. This makes overriding its version downstream way easier.

File: pkgs/development/python-modules/aiodiscover/default.nix [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+  patches = [
+    (fetchpatch {
+      name = "remove-entry_point.patch";
+      url = "https://github.com/bdraco/aiodiscover/pull/9/commits/4c497fb7d4c8685a78209c710e92e0bd17f46bb2.patch";
+      sha256 = "0py9alhg6qdncbn6a04mrnjhs4j19kg759dv69knpqzryikcfa63";
+    })
+  ];
+
+  nativeBuildInputs = [
+    pytest-runner
```
Comment (SuperSandro2000) [inline pkgs/development/python-modules/aiodiscover/default.nix:33]: Can we patch that out?

**Review by SuperSandro2000 — CHANGES_REQUESTED** [review]

**Comment by r-rmcgibbo** [comment]
Result of `nixpkgs-review pr 118176` at dd98712f run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>4 packages failed to build:</summary>
  <ul>
    <li><a href="https://gist.github.com/5b0a527d7be04516a93c4310e65d6348">python38Packages.aiodiscover</a></li>
    <li><a href="https://gist.github.com/e18f773c7d3ba942123617480b4b9683">python38Packages.async-dns</a></li>
    <li><a href="https://gist.github.com/aa59345fd50ce4b671ae993f3baa2e9d">python39Packages.aiodiscover</a></li>
    <li><a href="https://gist.github.com/950b739e0bfaa6fdb0797007001ed4ff">python39Packages.async-dns</a></li>
  </ul>
</details>
<details>
  <summary>2 packages built successfully:</summary>
  <ul>
    <li>python38Packages.pytest-raises</li>
    <li>python39Packages.pytest-raises</li>
  </ul>
</details>
<details>
  <summary>2 suggestions:</summary>
  <ul>
    <li>warning: <a href="https://github.com/jtojnar/nixpkgs-hammering/blob/master/explanations/python-include-tests.md">python-include-tests</a>

Consider adding a `checkPhase` for tests, or if not feasible, `pythonImportsCheck`.

Near pkgs/development/python-modules/aiodiscover/default.nix:47:0:
```
   |
47 |     description = "Python module to discover hosts via ARP and PTR lookup";
   | ^
```
</li>
    <li>warning: <a href="https://github.com/jtojnar/nixpkgs-hammering/blob/master/explanations/missing-patch-comment.md">missing-patch-comment</a>

Please add a comment on the line above, explaining the purpose of this patch.
Near pkgs/development/python-modules/aiodiscover/default.nix:25:5:
```
   |
25 |     (fetchpatch {
   |     ^
```
</li>
  </ul>
</details>

Note that build failures [may predate](https://github.com/nix-community/hydra-check) this PR, and could be nondeterministic or hardware dependent.
Please exercise your independent judgement.

**Comment by mweinelt** [comment]
Considering we have the `dhcp` component tests enabled, can you look into the requested changes quickly, so we can do the home-assistant release tomorrow without having to wait for this package?

File: pkgs/development/python-modules/async-dns/default.nix (hunk 1 above)
Comment (mweinelt) [reply pkgs/development/python-modules/async-dns/default.nix:23]: And add a remark to remove for async-dns>1.1.9

Comment (fabaff) [reply pkgs/development/python-modules/async-dns/default.nix:28]: Use patch for now.

File: pkgs/development/python-modules/pytest-raises/default.nix (hunk 2 above)
Comment (fabaff) [reply pkgs/development/python-modules/pytest-raises/default.nix]: Thanks

File: pkgs/development/python-modules/aiodiscover/default.nix (hunk 3 above)
Comment (fabaff) [reply pkgs/development/python-modules/aiodiscover/default.nix]: It's patched out now.

**Comment by fabaff** [comment]
Result of `nixpkgs-review pr 118176` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>6 packages built:</summary>
  <ul>
    <li>python38Packages.aiodiscover</li>
    <li>python38Packages.async-dns</li>
    <li>python38Packages.pytest-raises</li>
    <li>python39Packages.aiodiscover</li>
    <li>python39Packages.async-dns</li>
    <li>python39Packages.pytest-raises</li>
  </ul>
</details>

File: pkgs/development/python-modules/async-dns/default.nix [hunk 4]
```diff
@@ -0,0 +1,52 @@
...
+  pname = "async-dns";
+  version = "1.1.9";
+  disabled = pythonOlder "3.6";
+  format = "pyproject";
+
+  src = fetchFromGitHub {
+    owner = "gera2ld";
+    repo = "async_dns";
+    rev = "v${version}";
+    sha256 = "0hrpjgbssvr95z9jdnmx853r6h6mih0w138hz301ll94mi8zpbsg";
```
Comment (mweinelt) [inline pkgs/development/python-modules/async-dns/default.nix:20]: hash mismatch in fixed-output derivation '/nix/store/5pw32azgcikiv6gfrs4swzz5xhhcyqyd-source':
  wanted: sha256:0hrpjgbssvr95z9jdnmx853r6h6mih0w138hz301ll94mi8zpbsg
  got:    sha256:1z8j0s3dwcyavarhx41q75k1cmfzmwiqdh4svv3v15np26cywyag

**Comment by mweinelt** [comment]
Result of `nixpkgs-review pr 118176` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>6 packages built:</summary>
  <ul>
    <li>python38Packages.aiodiscover</li>
    <li>python38Packages.async-dns</li>
    <li>python38Packages.pytest-raises</li>
    <li>python39Packages.aiodiscover</li>
    <li>python39Packages.async-dns</li>
    <li>python39Packages.pytest-raises</li>
  </ul>
</details>

**Pull request merged**